bots/tests-invoke

```diff
@@ -170,6 +170,7 @@ class PullTask(object):
         try:
             if subprocess.call([ "git", "merge-base", "--is-ancestor", remote_base, "HEAD" ]) != 0:
                 sys.stderr.write("Rebasing onto " + remote_base + " ...\n")
+                subprocess.check_call([ "git", "reset", "HEAD" ])
```

Comment:
To clarify, this is only (potentially) affecting changed files in bots/, not somewhere else? I didn't think/play with this enough to be convinced that this is enough, or whether `--hard` is required; but if this fails, we should very clearly see this in fast-failing tests, so that's safe enough.

Reply:
It should only affect the index, no actual files in the working directory.